pkg/apis/options/legacy_options.go

```diff
@@ -0,0 +1,98 @@
+package options
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/logger"
+)
+
+type LegacyOptions struct {
+	// Legacy options related to upstream servers
+	LegacyFlushInterval                 time.Duration `flag:"flush-interval" cfg:"flush_interval"`
+	LegacyPassHostHeader                bool          `flag:"pass-host-header" cfg:"pass_host_header"`
+	LegacyProxyWebSockets               bool          `flag:"proxy-websockets" cfg:"proxy_websockets"`
+	LegacySSLUpstreamInsecureSkipVerify bool          `flag:"ssl-upstream-insecure-skip-verify" cfg:"ssl_upstream_insecure_skip_verify"`
+	LegacyUpstreams                     []string      `flag:"upstream" cfg:"upstreams"`
+
+	Options Options `cfg:",squash"`
+}
+
+func NewLegacyOptions() *LegacyOptions {
+	return &LegacyOptions{
+		LegacyPassHostHeader:  true,
+		LegacyProxyWebSockets: true,
+		LegacyFlushInterval:   time.Duration(1) * time.Second,
+
+		Options: *NewOptions(),
+	}
+}
+
+func (l *LegacyOptions) ToOptions() (*Options, error) {
```

Comment:
Do you have a longer term vision of this converting more types of legacy options into a the main `options.Options`?

Or is this more clear refactored into a method that's whole purpose is to generate an `Upstreams` object that can be set to `options.Options.Upstreams`?

Reply:
Yeah, this is the start of many of these conversions in theory. For example, I expect we will add something like a `convertLegacyHeaders` eventually that converts all of the `pass` and `set` options into a new structured system, as this has done for the upstreams